src/main/java/uk/gov/ons/ctp/response/casesvc/endpoint/CaseEndpoint.java

```diff
@@ -103,6 +104,29 @@ public CaseEndpoint(
     return ResponseEntity.ok(buildDetailedCaseDTO(caseObj, caseevents, iac));
   }
 
+  @RequestMapping(value = "/sampleunitids", method = RequestMethod.GET)
+  public ResponseEntity<CaseDetailsDTO[]> findCasesBySampleUnitIds(
+      @RequestParam(value = "sampleUnitId") final UUID[] sampleUnitIds,
+      @RequestParam(value = "caseevents", required = false) final boolean caseevents,
+      @RequestParam(value = "iac", required = false) final boolean iac)
+      throws CTPException {
+
+    List<UUID> ids = Arrays.asList(sampleUnitIds);
+
+    List<CaseDetailsDTO> cases =
+        ids.stream()
+            .filter(e -> caseService.findCaseBySampleUnitId(e) != null)
```

Comment:
We could possible use the in clause to do this as one query https://docs.spring.io/spring-data/jpa/docs/1.5.0.RC1/reference/html/repository-query-keywords.html

Reply:
We should also map first so we don't have to do the query twice per ID